packages/shell-api/src/field-level-encryption.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  classPlatforms,
+  hasAsyncChild,
+  returnsPromise,
+  ShellApiClass,
+  shellApiClassDefault
+} from './decorators';
+import { ReplPlatform } from '@mongosh/service-provider-core';
+import type { Document, BinaryType } from '@mongosh/service-provider-core';
+import Collection from './collection';
+import Cursor from './cursor';
+import { DeleteResult } from './result';
+import { assertArgsDefined, assertArgsType } from './helpers';
+import { asPrintable } from './enums';
+import { redactPassword } from '@mongosh/history';
+import type Mongo from './mongo';
+import { MongoshInvalidInputError } from '@mongosh/errors';
+
+/** Configuration options for using 'aws' as your KMS provider */
+declare interface awsKms {
+  aws: {
+    /** The access key used for the AWS KMS provider */
+    accessKeyId?: string;
+    /** The secret access key used for the AWS KMS provider */
+    secretAccessKey?: string;
+  };
+}
+
+/** Configuration options for using 'local' as your KMS provider */
+declare interface localKms {
+  local: {
+    /** The master key used to encrypt/decrypt data keys. A 96-byte long Buffer. */
+    key: BinaryType;
+  };
+}
+
+export interface ClientSideFieldLevelEncryptionOptions {
+  keyVaultClient?: any, /** This should be a Mongo class instance, but to avoid circular ref use any **/
+  keyVaultNamespace: string,
+  kmsProvider: awsKms | localKms,
+  schemaMap?: Document,
+  bypassAutoEncryption?: boolean;
+}
+
+@shellApiClassDefault
+@hasAsyncChild
+@classPlatforms([ ReplPlatform.CLI ] )
+export class ClientEncryption extends ShellApiClass {
+  public _mongo: Mongo;
+  public _libmongocrypt: any;
+
+  constructor(mongo: any) {
+    super();
+    this._mongo = mongo;
+    this._libmongocrypt = new mongo._serviceProvider.fle.ClientEncryption(
+      mongo._serviceProvider.getRawClient(),
+      {
+        options: this._mongo._fleOptions
```

Comment:
Should this be
```suggestion
        ...this._mongo._fleOptions
```
?

Reply:
Oh nice, yes it should absolutely be that

Reply:
I updated it and added a test